src/LayerPlan.cpp

```diff
@@ -1703,11 +1703,27 @@ void ExtruderPlan::processFanSpeedAndMinimalLayerTime(bool force_minimal_layer_t
 
 void LayerPlan::processFanSpeedAndMinimalLayerTime(Point starting_position)
 {
+    // the minimum layer time behaviour is only applied to the last extruder.
+    size_t max_extruder_nr = 0;
+    for (unsigned int extr_plan_idx = 0; extr_plan_idx < extruder_plans.size(); extr_plan_idx++)
+    {
+        max_extruder_nr = std::max(extruder_plans[extr_plan_idx].extruder_nr, max_extruder_nr);
+    }
+    // determine the time spend on the other extruder plans in this layer.
+    double time_other_extr_plans = 0;
+    for (unsigned int extr_plan_idx = 0; extr_plan_idx < extruder_plans.size(); extr_plan_idx++)
+    {
+        if (extr_plan_idx != max_extruder_nr)
+        {
+            time_other_extr_plans += extruder_plans[extr_plan_idx].computeNaiveTimeEstimates(starting_position).getTotalTime();
+        }
+    }
```

Comment:
```suggestion
    auto other_extr_plans = extruder_plans 
                          | ranges::views::filter([max_extruder_nr](const ExtruderPlan& extruder_plan) { return extruder_plan.extruder_nr != max_extruder_nr; })
                          | ranges::views::transform([starting_position](ExtruderPlan& extruder_plan) { return extruder_plan.computeNaiveTimeEstimates(starting_position).getTotalTime(); });
    const double time_other_extr_plans = ranges::accumulate(other_extr_plans, 0.0);
```

Don't forget to include at the top:
```
#include <range/v3/numeric/accumulate.hpp>
#include <range/v3/view/transform.hpp>
#include <range/v3/view/filter.hpp>
```

Reply:
outdated